x-pack/test/apm_api_integration/common/utils/expect_to_reject.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export async function expectToReject<T extends Error>(fn: () => Promise<any>): Promise<T> {
+  try {
+    await fn();
+  } catch (e) {
+    return e;
+  }
+  throw new Error(`Expected fn to throw`);
+}
```

Comment:
Thanks for adding this as a shared util. Will you replace the implementation in `x-pack/test/apm_api_integration/tests/settings/agent_keys/agent_keys.spec.ts` with this?